miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -733,11 +734,41 @@ public ModelAndView setupForm(@PathVariable Long libraryId, @PathVariable Long s
     }
   }
 
-  private final BulkTableBackend<Library, LibraryDto, Sample> libraryBulkBackend = new BulkTableBackend<Library, LibraryDto, Sample>(
-      "Libraries", "Samples", "library") {
+  private final BulkEditTableBackend<Library, LibraryDto> libraryBulkEditBackend = new BulkEditTableBackend<Library, LibraryDto>(
+      "library", LibraryDto.class, "Libraries") {
 
     @Override
-    protected LibraryDto packageDtoFromParent(Sample sample) {
+    protected Iterable<Library> load(List<Long> ids) throws IOException {
+      List<Library> results = libraryService.listByIdList(ids);
+      SampleClass sampleClass = null;
+      for (Library library : results) {
+        if (!isDetailedSampleEnabled()) {
+          // Do nothing about sample classes.
+        } else if (sampleClass == null) {
+          sampleClass = ((DetailedSample) library.getSample()).getSampleClass();
+        } else if (((DetailedSample) library.getSample()).getSampleClass().getId() != sampleClass.getId()) {
+          throw new IOException("Can only update libraries when samples all have the same class.");
```

Comment:
DNA and RNA have different LibraryDesigns available, and adding a LibraryDesign that's incompatible with the sample aliquot's class throws an error on save. I don't think we've had any complaints from our users about this, have we?